stix2/sources/filters.py

```diff
@@ -155,7 +116,53 @@ def apply_common_filters(stix_objs, query):
             yield stix_obj
 
 
-"""Base type filters"""
+def _check_filter(filter_, stix_obj):
+    """Evaluate a single filter against a single STIX 2.0 object.
+
+    Args:
+        filter_ (Filter): filter to match against
+        stix_obj: STIX object to apply the filter to
+
+    Returns:
+        True if the stix_obj matches the filter,
+        False if not.
+
+    """
+    if "." in filter_.field:
+        # For properties like granular_markings and external_references
+        # need to extract the first property from the string.
+        field = filter_.field.split(".")[0]
+    else:
+        field = filter_.field
+
+    if field not in stix_obj.keys():
+        # check filter "field" is in STIX object - if cant be
+        # applied to STIX object, STIX object is discarded
+        # (i.e. did not make it through the filter)
+        return False
+
+    if "." in filter_.field:
+        # Check embedded properties, from e.g. granular_markings or external_references
+        sub_field = filter_.field.split(".", 1)[1]
+        sub_filter = filter_._replace(field=sub_field)
+        if isinstance(stix_obj[field], list):
+            for elem in stix_obj[field]:
+                r = _check_filter(sub_filter, elem)
+                if r:
+                    return r
+            return False
+        else:
+            return _check_filter(sub_filter, stix_obj[field])
+    elif isinstance(stix_obj[field], list):
+        # Check each item in list property to see if it matches
+        for elem in stix_obj[field]:
+            r = _all_filter(filter_, elem)
+            if r:
+                return r
+        return False
+    else:
+        # Check if property matches
+        return _all_filter(filter_, stix_obj[field])
```

Comment:
maybe rename `_all_filter` to something else? Consider making it a method on the Filter class, since it touches two of the three data elements?

Reply:
I like this idea. Maybe call it `apply` or `apply_to_property`?

Which reminds me, I need to make a pass through this and make sure we call them `property` instead of `field`.